### DISCUSSION TITLE
Image issue: Signature is missing

### DISCUSSION THREAD

**Discussion by connor11528** [statamic/cms#4321] [Troubleshooting]
Hi there,

We’re running a Statamic blog inside my Laravel 8 app and are using a package artesaos/seotools to generate Open Graph images. It looks like this package creates its own URL for the image like the one below: 

https://www.employbl.com/img/asset/aW1hZ2VzL3Blb3BsZV9jb21wcmVzc2VkLnBuZw== 

However instead of getting an image the page is returning an error: “Signature is missing”.

![2C881F7D-C643-4882-AF47-143123E281D2](https://user-images.githubusercontent.com/3578908/134932684-1309a420-7b5f-4a91-8551-0fd52ee63fc4.png)

I’m having a bit of trouble understanding what this means and how to fix. Any ideas?

**Comment by jasonvarga** [comment]
The signature it's talking about is this:

https://github.com/statamic/statamic/blob/31e4b6861826a846ac868f28d93c0153655ebad2/config/statamic/assets.php#L19-L30

It's a security measure to prevent people from hitting a URL with different parameters and making your server generate tons of images.

The `{{ glide }}` tag would normally handle this. I'm not sure about the package you're using.

Can you link to the package? Or show how you're using it in Statamic?

**Reply by connor11528** [reply]
Hey for sure that's starting to make sense :) This is the package we're using: https://statamic.com/addons/cnj/seotamic. (I also an opened [an issue](https://github.com/cnj-digital/seotamic/issues/15) on the package repo.) In our codebase we drop in to the layout antlers file like:
```
{{ seotamic }}
```
Each article in our markdown blog posts we have some custom values like:
```
seotamic_title: custom
seotamic_title_prepend: true
seotamic_title_append: true
seotamic_meta_description: custom
seotamic_open_graph_title: custom
seotamic_open_graph_description: custom
seotamic_twitter_title: custom
seotamic_twitter_description: general
updated_by: e031d794-be15-4c4e-b254-7a5ad16c4afb
updated_at: 1629672301
hero_image: NY-tech-companies.jpg
hero_caption: 'Map of New York tech companies and startups.'
excerpt: 'This post is a breakdown of the current tech company and startup scene in NYC. It analyzes companies by HQ Location, Founded Year, Employee Count, Total Funding Amount and Industry.'
authors:
  - e031d794-be15-4c4e-b254-7a5ad16c4afb
seotamic_custom_title: 'Tech Companies and Startups in New York'
seotamic_custom_meta_description: 'This post is a breakdown of the current tech company and startup scene in NYC. It analyzes companies by HQ Location, Founded Year, Total Funding Amount and Industry.'
seotamic_custom_open_graph_title: 'Tech Companies and Startups in New York'
seotamic_custom_open_graph_description: 'This post is a breakdown of the current tech company and startup scene in NYC. It analyzes companies by HQ Location, Founded Year, Total Funding Amount and Industry.'
seotamic_custom_twitter_title: 'Tech Companies and Startups in New York'
seotamic_image: NY-tech-companies.jpg
```
Looks like the image manipulation happens here: https://github.com/cnj-digital/seotamic/blob/550c020017d9849ad14ea7266bf0be8f439c24f1/src/Tags/SeotamicTags.php#L215

Maybe I need to set up Glide or give permissions to generate the images in this limited context?
